liberapay/security/authentication.py

```diff
@@ -57,6 +63,41 @@ def sign_in_with_form_data(body, state):
             )
             if not p:
                 state['log-in.error'] = _("Bad username or password.")
+            else:
+                last_password_check = p.get_last_event_of_type('password-check')
+                if (not last_password_check) or utcnow() - last_password_check.ts > timedelta(days=30):
```

Comment:
I don't think we need to check every 30 days, once or twice per year should be enough, e.g. `days=180`.